src/tilesource.js

```diff
@@ -709,6 +709,166 @@ $.TileSource.prototype = {
+                success: function(request) {
+                    var blb;
+                    // Make the raw data into a blob.
+                    // BlobBuilder fallback adapted from
+                    // http://stackoverflow.com/questions/15293694/blob-constructor-browser-compatibility
+                    try {
+                        blb = new window.Blob([request.response]);
+                    } catch (e) {
+                        var BlobBuilder = (
+                            window.BlobBuilder ||
+                            window.WebKitBlobBuilder ||
+                            window.MozBlobBuilder ||
+                            window.MSBlobBuilder
+                        );
+                        if (e.name === 'TypeError' && BlobBuilder) {
+                            var bb = new BlobBuilder();
+                            bb.append(request.response);
+                            blb = bb.getBlob();
+                        }
+                    }
+                    // If the blob is empty for some reason consider the image load a failure.
+                    if (blb.size === 0) {
+                        context.finish(false, "Empty image response.");
+                    } else {
+                        // Create a URL for the blob data and make it the source of the image object.
+                        // This will still trigger Image.onload to indicate a successful tile load.
+                        context.image.src = (window.URL || window.webkitURL).createObjectURL(blb);
+                    }
+                },
+                error: function(request) {
+                    context.finish(false, "Image load aborted - XHR error");
+                }
+            });
+        } else {
+            if (context.crossOriginPolicy !== false) {
+                context.image.crossOrigin = context.crossOriginPolicy;
+            }
+
+            context.image.src = context.src;
+        }
+    },
+
+    /**
+     * @param {object} context job, the same object as with downloadTileStart(..)
+     * @param successful true if successful
+     * @return {null|*} null to indicate missing data or data object
+     *  for example, can return default value if the request was unsuccessful such as default error image
+     */
+    downloadTileFinish: function (context, successful) {
+        if (!context.image) {
+            return null;
+        }
+        context.image.onload = context.image.onerror = context.image.onabort = null;
+        if (!successful) {
+            return null;
+        }
+        return context.image;
+    },
+
+    /**
+     * Create cache object from the result of the download process
+     * @param {Tile} tile instance the cache was created with
+     * @param {object} cacheObject context cache object
+     * @param {*} data the result of downloadTileFinish() function
+     */
+    createTileCache: function(cacheObject, data, tile) {
+        cacheObject._data = data;
```

Comment:
Does it always need to be `_data`? If so, that should be documented here. Are there any other requirements?

Reply:
Yes, I will document this more. No, there are no requirements, as long as it's working for you. You get the data, you decide what to do with it. There is actually an implicit requirement of storage, otherwise, it will not work, worth documenting too. Good point.